### PR TITLE
feat(VSelectionControl): create new component

### PR DESCRIPTION
**Pull request by johnleider**
## TODO:
- [x] RadioGroup still exists but hasn't been updated, is there a recommended way to associate a label and validation to a set of radios now?- 
- [x] Checkbox/switch don't work with v-model
- [x] Missing true-value/false-value
- [x] Commented out or unused code still needs to be removed
- [x] In v2 you could click on the entire component to toggle it, now you have to hit the label or icon directly instead. This is especially annoying on switches because the target area can move.
- [x] Switch density is still busted, the current transform values only work if the thumb is 40px wide
- [x] Checkbox doesn't work with value properly, the icon changes color but doesn't appear checked because it's using vue's checkbox model behaviour (true/false ignoring value)
- [x] Multiple checkbox (array model) sets the model correctly but always appears unchecked
- [x] Unit tests
- [x] E2E tests

## Markup:

<details>

```vue
<template>
  <v-app>
    <div class="ma-4 pa-4">
      <!-- <v-checkbox label="Checkbox 1" disabled /> -->
      <v-switch label="Switch 2" :error="error" :disabled="disabled" color="primary" />
      <v-checkbox label="Checkbox 2" :error="error" :disabled="disabled" color="primary" />
      <v-radio
        v-model="radio"
        label="radio 2"
        :error="error"
        :disabled="disabled"
        value="see"
        color="primary"
      />

      <div class="my-8" />
      <v-text-field label="Text-field 2" :error="error" :disabled="disabled" color="primary" />

      <br>
      <br>
      <br>
      <br>

      <v-btn @click="error = !error">Error: {{ error }}</v-btn>
      <v-btn @click="disabled = !disabled">Disabled: {{ disabled }}</v-btn>

      <!--<v-radio v-model="radio" label="Radio 1" value="one" />
      <v-radio v-model="radio" label="Radio 2" value="two" />
      <v-radio v-model="radio" label="Radio 3" value="three" /> -->

    </div>
  </v-app>
</template>

<script>
  export default {
    data: () => ({
      radio: null,
      error: false,
      disabled: false,
    }),
  }
</script>

```
</details>

<details>

```vue
<template>
  <v-app>
    <v-container fluid>
      <v-row>
        <v-col>
          <v-defaults-provider :defaults="{ global: { density }}">
            <!-- <v-checkbox label="Checkbox 1" disabled /> -->
            <v-checkbox v-model="indeterminate" label="indeterminate" />
            <v-switch
              v-model="bool"
              v-model:indeterminate="indeterminate"
              label="On"
              :error="error"
              :disabled="disabled"
              color="primary"
              :true-value="true"
              :false-value="undefined"
            />
            <v-checkbox
              v-model="bool"
              v-model:indeterminate="indeterminate"
              label="Checkbox 2"
              :error="error"
              :disabled="disabled"
              color="primary"
              value="checked"
            />
            <v-radio
              v-model="radio"
              label="radio 2"
              :error="error"
              :disabled="disabled"
              value="see"
              color="primary"
              name="1"
            />

            <v-checkbox v-model="multiple" value="a" label="a" />
            <v-checkbox v-model="multiple" value="b" label="b" />
            <v-checkbox v-model="multiple" value="c" label="c" />

            <input v-model="multiple" type="checkbox" value="a">
            <input v-model="multiple" type="checkbox" value="b">
            <input v-model="multiple" type="checkbox" value="c">

            <div class="my-8" />
            <v-text-field label="Text-field 2" :error="error" :disabled="disabled" color="primary" />

            <br>
            <br>
            <br>
            <br>

            <v-btn @click="error = !error">Error: {{ error }}</v-btn>
            <v-btn @click="disabled = !disabled">Disabled: {{ disabled }}</v-btn>

            <v-radio-group v-model="radio2">
              <v-radio label="Radio 1" value="one" />
              <v-radio label="Radio 2" value="two" />
              <v-radio label="Radio 3" value="three" />
            </v-radio-group>

          </v-defaults-provider>
        </v-col>
        <v-col cols="2">
          <pre>{{ format({ multiple, radio, radio2, bool }) }}</pre>
          <input v-model="density" type="checkbox" true-value="default">
          <input v-model="density" type="checkbox" true-value="comfortable">
          <input v-model="density" type="checkbox" true-value="compact">
        </v-col>
      </v-row>
    </v-container>
  </v-app>
</template>

<script>
  export default {
    data: () => ({
      indeterminate: true,
      density: 'default',
      bool: false,
      multiple: [],
      radio: null,
      radio2: null,
      error: false,
      disabled: false,
    }),
    methods: {
      format (val) {
        return JSON.stringify(val, (k, v) => {
          return typeof v === 'function' ? '[Function]'
            : typeof v === 'undefined' ? '[Undefined]'
              : v
        }, 2)
      },
    },
  }
</script>
```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [ ] The PR title is no longer than 64 characters.
- [ ] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [ ] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VSwitch/VSwitch.sass [hunk 1]
```diff
@@ -1,132 +1,137 @@
...
+  border-radius: $switch-track-border-radius
+  height: $switch-track-height
+  left: $switch-track-x
+  opacity: $switch-track-opacity
+  pointer-events: none
+  position: absolute
+  right: $switch-track-x
+  top: 50%
+  transform: translateY(-50%)
+  width: $switch-track-width
```
Comment (KaelWD) [inline packages/vuetify/src/components/VSwitch/VSwitch.sass:28]: This should be `--v-selection-control-size - $switch-track-x * 2`

File: packages/vuetify/src/components/VSwitch/VSwitch.sass [hunk 2]
```diff
@@ -1,132 +1,137 @@
...
-
-    &.v-input--is-dirty
-      .v-input--selection-controls__ripple,
-      .v-input--switch__thumb
-        +ltr()
-          transform: translate($switch-dirty-offset-x, 0) !important
-
-        +rtl()
-          transform: translate(-$switch-dirty-offset-x - 6px, 0) !important
+    transition: .15s settings.$standard-easing
```
Comment (KaelWD) [inline packages/vuetify/src/components/VSwitch/VSwitch.sass:44]: Transition specific properties

**Review by KaelWD — COMMENTED** [review]
- RadioGroup still exists but hasn't been updated, is there a recommended way to associate a label and validation to a set of radios now?
  - This also means you have to use `name` on radios so they don't all end up in one group
- Checkbox/switch don't work with v-model
- Missing true-value/false-value
- VInput prepend/append elements always exist even if unused
- Commented out or unused code still needs to be removed
- In v2 you could click on the entire component to toggle it, now you have to hit the label or icon directly instead. This is especially annoying on switches because the target area can move.

File: packages/vuetify/src/components/VField/VField.sass [hunk 3]
```diff
@@ -364,30 +353,10 @@
...
     color: $field-error-color
-
-    .v-icon,
-    .v-messages,
-    .v-counter
-      color: $field-error-color
-
-    .v-field__outline::after
-      transform: scaleX(1)
+    transform: scaleX(1)
```
[the remark this answers is not in the document]
Comment (johnleider) [reply packages/vuetify/src/components/VField/VField.sass:359]: Not according to the [spec](https://material.io/components/text-fields)

image

File: packages/vuetify/src/components/VLabel/VLabel.sass [hunk 4]
```diff
@@ -1,14 +1,25 @@
...
-  &--is-disabled
-    color: map-deep-get($material, 'text', 'disabled')
+@forward './variables'
+@use '../../styles/settings'
+@use '../../styles/tools'
+@use './variables' as *
 
 .v-label
+  color: $label-color
+  contain: layout paint
```
[the remark this answers is not in the document]
Comment (johnleider) [reply packages/vuetify/src/components/VLabel/VLabel.sass:8]: color is still desirable in v-label

File: packages/vuetify/src/components/VSelectionControl/VSelectionControl.sass [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    .v-icon
+      color: rgb(var(--v-theme-error))
+
+  + .v-input__details
+    padding: 0 12px
+
+  @at-root
+    @include tools.density('v-selection-control', $selection-control-density) using ($modifier)
+      .v-selection-control__input
+        min-width: $selection-control-size + $modifier
```
[the remark this answers is not in the document]
Comment (johnleider) [reply packages/vuetify/src/components/VSelectionControl/VSelectionControl.sass:41]: Display issues without it. 

image

File: packages/vuetify/src/components/VSwitch/VSwitch.sass (hunk 1 above)
Comment (johnleider) [reply packages/vuetify/src/components/VSwitch/VSwitch.sass:28]: Need your help with this one, not understanding what this changes.

File: packages/vuetify/src/components/VField/VField.sass (hunk 3 above)
[the remark this answers is not in the document]
Comment (KaelWD) [reply packages/vuetify/src/components/VField/VField.sass:367]: That proves my point though? The bottom bar is only 1px in your screenshot

File: packages/vuetify/src/components/VSelectionControl/VSelectionControl.sass (hunk 5 above)
[the remark this answers is not in the document]
Comment (KaelWD) [reply packages/vuetify/src/components/VSelectionControl/VSelectionControl.sass:41]: This is caused by `flex: 0 1` (`0 1 0`) allowing the element to shrink.

File: packages/vuetify/src/components/VSwitch/_variables.scss [hunk 6]
```diff
@@ -1,7 +1,9 @@
-@import '../../styles/styles.sass';
-
+$switch-control-input-transform: translateX(calc(#{var(--v-selection-control-size)} * -1)) !default;
+$switch-dirty-control-input-transform: translateX(calc(#{var(--v-selection-control-size)})) !default;
```
[the remark this answers is not in the document]
Comment (KaelWD) [reply packages/vuetify/src/components/VSwitch/_variables.scss]: This is because `--v-selection-control-size` is also using interpolation incorrectly

**Comment by KaelWD** [comment]
- Switch density is still busted, the current transform values only work if the thumb is 40px wide
- Checkbox doesn't work with value properly, the icon changes color but doesn't appear checked because it's using vue's checkbox model behaviour (true/false ignoring value)
- Multiple checkbox (array model) sets the model correctly but always appears unchecked

File: packages/vuetify/src/components/VField/VField.sass (hunk 3 above)
[the remark this answers is not in the document]
Comment (johnleider) [reply packages/vuetify/src/components/VField/VField.sass:369]: You're right, will resolve.

**Pull request merged**